File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    
    #로또 번호 정리
    win = dict()
    for i in range(6):
        win[win_nums[i]] = True

    #순위
    level = dict()
    for i in range(7):
        if i == 0 or i == 1: #한개 또는 아예 못 맞춘 경우
            level[i] = 6
        else:
            level[i] = 7-i

    #최저 경우 구하기, 0의 개수 구하기
    zero = 0 #0의 개수
    equal = 0 #맞힌 로또 번호의 개수
    for i in range(6):
        if lottos[i] in win:
            equal += 1
        elif lottos[i] == 0:
            zero += 1

    #가능한 범위
    pos = zero+equal
    answer = [level[pos],level[equal]]
    
    return answer

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
//10:47
//당첨 번호순으로 순위가 정해짐
//당첨이 가능했던 최고 및 최저 순위를 알고 싶다 -> Map을 통해 맞춘 양에 따라 순위 값 구해주면 될듯

import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = {0, 0};
        
        //score 초기화
        HashMap<Integer, Integer> score = new HashMap<Integer, Integer>();
        
        int[] a = {1,2,3,4,5,6,6};
        int[] b = {6,5,4,3,2,1,0};
        
        for (int i = 0; i<a.length; i++) {
            score.put(b[i], a[i]);
        }
        
        //Set으로 당첨 번호 초기화, 시간복잡도를 위해 초기화
        HashSet<Integer> nums = new HashSet<Integer>();
        for (int i : win_nums) {
            nums.add(i);
        }
        
        int count = 0; //0의 수
        int right = 0; //맞은 수
        
        for (int i : lottos) {
            
            if (i == 0) {
                count++;
            }
            
            if (nums.contains(i)) {
                right++;
            }
        }
        
        answer[0] = score.get(count+right);
        answer[1] = score.get(right);
        
        return answer;
    }
}